edx_sga/sga.py

```
@@ -0,0 +1,530 @@
+                    'module_id': module.id,
+                    'student_id': student.student_id,
+                    'submission_id': submission['uuid'],
+                    'username': module.student.username,
+                    'fullname': module.student.profile.name,
+                    'filename': submission['answer']["filename"],
+                    'timestamp': submission['created_at'].strftime(
+                        DateTime.DATETIME_FORMAT
+                    ),
+                    'score': score,
+                    'approved': approved,
+                    'needs_approval': instructor and needs_approval,
+                    'may_grade': instructor or not approved,
+                    'annotated': state.get("annotated_filename"),
+                    'comment': state.get("comment", ''),
+                }
+
+        return {
+            'assignments': list(get_student_data()),
+            'max_score': self.max_score(),
+        }
+
+    def studio_view(self, context=None):
+        try:
+            cls = type(self)
+
+            def none_to_empty(x):
+                return x if x is not None else ''
+            edit_fields = (
+                (field, none_to_empty(getattr(self, field.name)), validator)
+                for field, validator in (
+                    (cls.display_name, 'string'),
+                    (cls.points, 'number'),
+                    (cls.weight, 'number'))
+            )
+
+            context = {
+                'fields': edit_fields
+            }
+            fragment = Fragment()
+            fragment.add_content(
+                render_template(
+                    'templates/staff_graded_assignment/edit.html',
+                    context
+                )
+            )
+            fragment.add_javascript(_resource("static/js/src/studio.js"))
+            fragment.initialize_js('StaffGradedAssignmentXBlock')
+            return fragment
+        except:  # pragma: NO COVER
```

Does the studio_view function somehow swallow exceptions? That sounds like a problem we should resolve. Is the idea here that we'll at least log it as error at the XBlock level?